src/NHibernate/Cfg/Environment.cs

```diff
@@ -396,6 +397,30 @@ public static string Version
 		/// </summary>
 		public const string MultiTenancyConnectionProvider = "multi_tenancy.connection_provider";
 
+		/// <summary>
+		/// The maximum number of entries including:
+		/// <list>
+		/// <item>
+		///	<see cref="HQLQueryPlan"/>
+		/// </item>
+		/// <item>
+		/// <see cref="NativeSQLQueryPlan"/>
+		/// </item>
+		/// <item>
+		/// <see cref="FilterQueryPlan"/>
+		/// </item>
+		/// </list>
+		/// 
+		/// maintained by <see cref="QueryPlanCache"/>. Default is 128.
+		/// </summary>
+		public const string QueryPlanCacheMaxSize = "query.plan_cache_max_size";
+
+		/// <summary>
+		/// The maximum number of <see cref="ParameterMetadata"/> maintained
+		/// by <see cref="QueryPlanCache"/>. Default is 128.
+		/// </summary>
+		public const string QueryPlanCacheParameterMetadataMaxSize = "query.plan_parameter_metadata_max_size";
```

Comment:
does it need to be added to the configuration schema?

Reply:
Good catch! Done.